src/c/secrets-vault.c

```diff
@@ -299,6 +326,56 @@ static void vault_releaseregtoken (void *impl)
   pthread_mutex_unlock (&vault->mtx);
 }
 
+static iot_data_t * vault_requestjwt (void *impl)
+{
+  iot_data_t *result = NULL;
+  vault_impl_t *vault = (vault_impl_t *)impl;
+  
+  iot_data_t *reply = vault_rest_get (vault, vault->jwtissueurl);
+  if (reply)
+  {
+    const iot_data_t *d = iot_data_string_map_get (reply, "data");
+    if (d)
+    {
+      const iot_data_t *t = iot_data_string_map_get (d, "token");
+      if (t)
+      {
+        result = iot_data_copy (t);
+      }
+    }
+  }
+  if (result == NULL)
+  {
+    iot_log_error (vault->lc, "vault: get JWT request failed");
+    result = iot_data_alloc_map (IOT_DATA_STRING);
+  }
+  iot_data_free (reply);
+  return result;
+}
+
+static bool vault_isjwtvalid (void *impl, const char *jwt)
+{
+  bool result = false;
+  vault_impl_t *vault = (vault_impl_t *)impl;
+
+  JSON_Value *jval = json_value_init_object ();
```

Comment:
Use `iot_data_alloc_map` / `iot_data_string_map_add` / `iot_data_to_json` instead
(The use of `parson.h` code dates back to when we didn't have JSON support in `iot/data.h`. Plan at the moment is to phase it out as and when the opportunity arises)

Reply:
@iain-anderson 

Please check my work: https://github.com/edgexfoundry/device-sdk-c/pull/442/commits/e4c511e74d56997ceda3718f127bf79c27b87423

I don't know this library very well, so keep an eye out for memory leaks!

Reply:
Looks ok. I run stuff under `valgrind` a lot while working on this, it's very useful for picking up memory problems